airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/LauncherWorker.java

```diff
@@ -117,12 +119,30 @@ public OUTPUT run(INPUT input, Path jobRoot) throws WorkerException {
             containerOrchestratorConfig.containerOrchestratorImage(),
             containerOrchestratorConfig.googleApplicationCredentials());
 
+        cancellationCallback.set(() -> {
+          // When cancelled, try to set to true.
+          // Only proceed if value was previously false, so we only have one cancellation going. at a time
+          if (!isCanceled.getAndSet(true)) {
+            log.info("Trying to cancel async pod process.");
+            process.destroy();
+          }
+        });
+
+        // only kill running pods and create process for the first run for an attempt
```

Comment:
is this comment accurate? is it about the first run? this is my understanding (below). am I misunderstanding?

```suggestion
        // only kill running pods and create process if it is not already running.
```